src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionBuilder.cs

```diff
@@ -96,6 +96,37 @@ private static bool ShouldExcludeEntryStu3(string resourceType, string searchPar
             return new BundleWrapper(modelInfoProvider.ToTypedElement(rawResource));
         }
 
+        internal static SearchParameterInfo GetOrCreateSearchParameterInfo(SearchParameterWrapper searchParameter, IDictionary<Uri, SearchParameterInfo> uriDictionary = null)
```

Comment:
>GetOrCreateSearchParameterInfo [](start = 44, length = 30)

From how it's used, I would break it into two things:
a) something what creates SearchParameterInfo out of SearchParameterWrapper. (If it's possbile I would just add constructor to SearchParameterInfo which accepts wrapper
b) private function, or maybe just don't have function at all.. 
```
uriDictionary.TryGetValue(new Uri(searchParameter.Url), out var spi) ? spi : new SearchParameterInfo(searchParameter)
```
doesn't like lot of code to be honest...

Reply:
Constructor added, I like that change as well.